rclcpp/include/rclcpp/clock.hpp

```diff
@@ -137,6 +137,57 @@ class Clock
     Duration rel_time,
     Context::SharedPtr context = contexts::get_global_default_context());
 
+  /**
+   * Check if the clock is valid.
+   *
+   * A valid clock is a clock that has a non-zero time.
+   * An invalid clock is an uninitialized clock (i.e. not rcl_clock_valid) or a clock with an
+   * uninitialized time (i.e. with zero time.)
+   *
+   * Note:
+   *     This notion of validity is different from the rcl notion of validity (rcl_clock_valid)!
+   *     A clock that is rcl_clock_valid is a clock that can be used to get a time (i.e. not
+   *     NULL, not RCL_CLOCK_UNINITIALIZED, or with a missing get_now() method.)
+   *
+   *     A clock that is valid in the sense codified by this method is a clock with non-zero time.
+   *     Consequently, a clock that is not rcl_clock_valid can never become valid, since it cannot
+   *     be used to obtain a time.
+   *
+   * \return true if clock was or became valid
+   */
+  RCLCPP_PUBLIC
+  bool
+  is_valid();
+
+  /**
+   * Wait for clock to become valid.
+   *
+   * \param context the context to wait in
+   * \return true if clock was or became valid
+   * \throws std::runtime_error if the context is invalid or clock is not rcl_clock_valid
+   */
+  RCLCPP_PUBLIC
+  bool
+  wait_for_valid(Context::SharedPtr context = contexts::get_global_default_context());
```

Comment:
what's the use case of wait_for_valid()?
It seems like a strange feature to me

Reply:
It'll block till the first valid time. In most cases this will be waiting for sim to start (and time to march forward)